### PR TITLE
feat(jira): run connector checks after the sync, fix task-history ordering, record silently cleared values, key issues on their immutable id

### PR DESCRIPTION
**Pull request by mitasovr**
Four changes, each standing on its own. What they have in common: none of them is observable on controlled inputs — they only appear when history is compared with the state a real issue holds.

## 1. A connector's own checks now run after its sync

`dbt test` runs in exactly one place in a cluster: the scheduled `data_quality` catalog. That catalog is install-wide, so it must read silver/gold only — a check touching one connector's bronze would fail on a tenant where that connector is absent. The Jira checks read that connector's bronze and staging, so they could not join it, and therefore ran **nowhere** outside CI. One of them states in its own comment that it "warns in the nightly run", which was never true.

This adds a `connector_quality` tag for checks that need a connector's own tables, and a `connector-checks` step at the end of that connector's pipeline, where the tables exist precisely because the pipeline built them. Each such check also carries its connector's tag, and the selector is the INTERSECTION of the two — a space-separated selector is a union, which would run every connector's checks on every connector's sync.

The step never fails the sync, by construction: bronze has landed and silver is rebuilt by the time it runs, so failing the workflow would undo nothing and would only turn a data condition into a red pipeline nobody can act on. It does refuse an empty selector, because `dbt test --select ""` matches nothing and exits 0 — a silent no-op is worse than an error.

## 2. Findings reach monitoring

Both Jira checks gain `store_failures` and the `meta` block the catalog convention expects, so each finding is emitted as a structured line carrying its `tier`. `severity` stays `warn`: these report a condition of the SOURCE, which no change here can repair, and a gate that can never go green stops being read. The finding emitter learns the new tag.

Their `remediation` explains how to tell the three causes apart, because on real data they look identical: a genuine parsing fault; a source that changed a value without recording an event; and a value whose identifier is not stable across the two sides. It also names the cheap discriminator — compare the raw changelog value with the raw issue JSON before suspecting the transformation.

## 3. Current task state is read by the full ordering key

`task_issue_state` resolved current state with `argMax(..., (event_at, _version))`. `_version` is a build stamp shared by every row of one build, so on two rows of the same instant the key ties and the winner is whatever the query plan produced.

This is not a corner case. A `synthetic_initial` row and a `changelog` row routinely share an issue's creation timestamp — that happens to every issue whose first event landed when it was created — and by `_seq` alone the initial row sorts AFTER the event, so the field reads as the empty state it had before. Two changelog rows can also share a millisecond.

The key is now `(event_at, task_event_rank(event_kind), _seq, toUInt64OrZero(event_id))`: kind first, because an initial row is by definition the state before any event; then `_seq`, which orders the initial rows among themselves; then the event id compared NUMERICALLY, since as text `'101'` sorts before `'99'`. The round-trip check already ordered this way — gold now agrees with it.

`jira_event_rank` becomes `task_event_rank` in the shared macro directory: the class it orders is written by every task-tracking connector, so the rank was not Jira's to own.

## 4. A value the issue cleared without an event is recorded

Jira lets a value disappear without a changelog entry — a link removed from the other side of the pair, an automation writing a read-only field, a list emptied in bulk. The journal keeps the last value it was told about, and the issue's own state disagrees with it for as long as the issue lives.

`retired_field` does not cover this: it fires when the key is ABSENT, meaning the field left the issue's context. Here the key is present and empty, which per §6 means the field still applies and is unset.

The new `snapshot_diff` kind fires only when all three hold: the field's events end on a value, the snapshot has no row, and the key IS in the issue JSON. It is dated by the observation, not by a guess at when the clearing happened, and its `event_id` is constant per (issue, field), so re-observing restamps one row instead of accumulating one per sync. It is its own kind so that a consumer counting real history can exclude it, and so the share of state recovered by observation stays measurable.

**The oracle keeps its sight.** A `snapshot_diff` row is the snapshot's value written back, so if the round-trip check counted it, the check would compare the snapshot with itself and report agreement regardless of what the events say — going blind on exactly the pairs it should be loudest about. It therefore ignores that kind, and a pair needing such a row still fails it. The scenario test asserts both halves.

Nothing reaches silver from here yet: the class is fed by the Rust pass-through until the cutover, and the silver `event_kind` enum will gain this kind together with `retired_field` and `unclassified_field` at that point.

## 5. An issue moved between projects no longer exists twice

`unique_key` on `jira_issue` and `jira_issue_keys` was built from the issue's "PROJ-123" key, and Jira changes that key when an issue moves between projects. Bronze replaces rows by `unique_key`, so a moved issue stopped replacing itself: the new key inserted a second row while the old one stayed behind with whatever payload it last held. Downstream the issue then appeared twice — once carrying its real history, once as a ghost built from a stale snapshot.

The connector now emits the immutable numeric id. Rows written before this keep their old key: `unique_key` is the table's sort key and ClickHouse refuses to `UPDATE` one, so no migration can rewrite them in place, and rebuilding a table of this size for a column nothing reads afterwards would not earn its cost. Every dedup over `jira_issue` instead picks its winner by `(source_id, jira_id)` — the snapshot, the derived journal and the long-text side table — so the stale copy is simply never chosen.

`id_readable` keeps the mutable key; the rename is a fact about the issue, and that is the column readers use for it.

The descriptor goes to **6.0.0**. Major, because the key an existing row is stored under changes — and the full refresh a major dispatches also clears anything derived while both copies were visible.

## 6. A cleared option list is no longer recorded as an option named `[]`

For `option_array` the display side stands in when the id side is missing, and the test for "missing" was an empty id list. Clearing the field produces exactly that, and the display of an empty bracketed list renders as the literal text `[]` — so the journal recorded the field as holding one option called `[]` instead of holding nothing. The fallback now triggers only when the id side is ABSENT; present and empty is a real value, meaning the field was cleared.

Two fixtures cover it, and both fail against the previous macro.

## Checks

- `src/ingestion/dbt/tests/jira/transform` — 73/73 pass.
- The chart renders with a real environment values file; the new step and its selector appear in the rendered pipeline.
- Both defects above were found by comparing the derived journal with the state real issues hold, and both fixes were verified the same way before and after.

## Summary by CodeRabbit

- **New Features**
  - Connector-specific data quality checks now run automatically after ingestion transforms.
  - Connector check results are reported as structured data-quality findings.
  - Jira history now records fields cleared in source snapshots.
  - Jira records use immutable issue IDs for more consistent tracking across key changes.

- **Bug Fixes**
  - Improved event ordering when Jira events share timestamps.
  - Corrected handling of empty Jira option values and snapshot-derived history.
  - Updated validation to recognize and verify snapshot-derived history changes.

- **Documentation**
  - Updated Jira field-history guidance to reflect revised event-ordering terminology.

### CODE REVIEW

File: charts/insight/templates/ingestion/ingestion-pipeline.yaml [hunk 1]
```diff
@@ -212,6 +212,38 @@ spec:
...
+          # Never fails the sync — see connector-checks.yaml.
+          - name: connector-checks
+            depends: transform-jira-silver.Succeeded || transform-legacy.Succeeded
+            templateRef:
+              name: connector-checks
+              template: run
+            arguments:
+              parameters:
+                - name: dbt_select
+                  value: {{ `"tag:connector_quality tag:{{inputs.parameters.data_source}}"` }}
```
[the remark this answers is not in the document]
Comment (mitasovr) [reply charts/insight/templates/ingestion/ingestion-pipeline.yaml]: Fixed in 82552ac — and the finding was understated. No test in the repository carries a bare connector tag, so the second half of that union (`tag:jira`) selected nothing at all; the selector collapsed to "every connector-quality check, on every connector's sync".

Both checks now carry `jira` alongside `connector_quality`, and the selector is the intersection. Verified with `dbt ls`: `tag:connector_quality,tag:jira` resolves to exactly the two checks, and the same selector with another connector resolves to nothing.

File: charts/insight/templates/ingestion/connector-checks.yaml [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+      # as "$ENV_VAR", never spliced into shell, Python or YAML source.
+      inputs:
+        parameters:
+          - name: dbt_select
+          - name: toolbox_image
+            default: {{ required "ingestion.toolboxImage is required when ingestion.templates.enabled=true" .Values.ingestion.toolboxImage | quote }}
+          - name: clickhouse_host
+            default: {{ include "insight.clickhouse.fqdn" . | quote }}
+          - name: clickhouse_port
+            default: {{ include "insight.clickhouse.port" . | quote }}
```
[the remark this answers is not in the document]
Comment (mitasovr) [reply charts/insight/templates/ingestion/connector-checks.yaml:42]: Not taking this one here, for two reasons.

It is not introduced by this PR: `secure: False` lives in `dbt_profiles.py`, which this PR does not touch, and every ingestion step that talks to ClickHouse already goes through it — `dbt-run` and `data-quality-test` included. Flipping it in a PR about check scheduling would be an unrelated, install-wide change.

And it cannot be flipped blind: `secure: True` makes the client speak TLS, so it fails outright wherever ClickHouse does not terminate TLS on the configured port. Making that safe means giving the profile a real setting driven by chart values, defaulting to the deployment's actual endpoint, plus certificate trust for self-signed setups — a deliberate change with its own rollout, not a one-line edit.

Worth doing, worth its own issue.

**Review by cyberantonz — APPROVED** [review]
